testlib/resources.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package testlib
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-server/utils"
+	"github.com/mattermost/mattermost-server/utils/fileutils"
+)
+
+const (
+	resourceTypeFile = iota
+	resourceTypeFolder
+)
+
+const (
+	actionCopy = iota
+	actionSymlink
+)
+
+type testResourceDetails struct {
+	src     string
+	dest    string
+	resType int8
+	action  int8
+}
+
+var testResourcesToSetup = []testResourceDetails{
+	{"config/timezones.json", "config/timezones.json", resourceTypeFile, actionCopy},
+	{"i18n", "i18n", resourceTypeFolder, actionSymlink},
+	{"templates", "templates", resourceTypeFolder, actionSymlink},
+	{"tests", "tests", resourceTypeFolder, actionSymlink},
+	{"fonts", "fonts", resourceTypeFolder, actionSymlink},
+	{"utils/policies-roles-mapping.json", "utils/policies-roles-mapping.json", resourceTypeFile, actionSymlink},
+}
+
+func resolveResourcesPath() {
+	var srcPath string
+	var found bool
+
+	// Finding resources and setting full path to source to be used for further processing
+	for i, testResource := range testResourcesToSetup {
+		if testResource.resType == resourceTypeFile {
+			srcPath = fileutils.FindFile(testResource.src)
+			if srcPath == "" {
+				panic(fmt.Sprintf("Failed to find file %s", testResource.src))
+			}
+
+			testResourcesToSetup[i].src = srcPath
+		} else if testResource.resType == resourceTypeFolder {
+			srcPath, found = fileutils.FindDir(testResource.src)
+			if found == false {
+				panic(fmt.Sprintf("Failed to find folder %s", testResource.src))
+			}
+
+			testResourcesToSetup[i].src = srcPath
+		} else {
+			panic(fmt.Sprintf("Invalid resource type: %d", testResource.resType))
+		}
+	}
+}
+
+func SetupTestResources() (string, error) {
+	resolveResourcesPath()
```

Comment:
Can we avoid working with globals? Seems like `testResourcesToSetup` could just be declared inside `resolveResourcesPath ` and returned instead of being global in this context.